apps/b2b/pages/auth/index.tsx

```diff
@@ -0,0 +1,26 @@
+import { useSession } from 'next-auth/react'
+import { useRouter } from 'next/router'
+
+export const REDIRECT_MAP: Record<Role, string> = {
+  USER: '/dashboard',
+  PUBLIC: '/',
+}
+
+const Auth = () => {
+  const { push } = useRouter()
+  const { status, data } = useSession()
+
+  if (status === 'loading') return null
+
+  if (status === 'authenticated') {
+    push({ pathname: REDIRECT_MAP[data.user.role] })
+  }
+
+  if (status === 'unauthenticated') {
+    push('/')
+  }
+
+  return null
+}
```

Comment:
this doesn't seem to be used anywhere?

Reply:
I needed to use this for the signIn callbackUrl to redirect the "user" to the proper portal

Reply:
I also want to checkout if it's better to use ``signIn redirect: false`` [documentation](https://next-auth.js.org/getting-started/client#using-the-redirect-false-option-1)